src/platform_impl/web/event_loop/runner.rs

```diff
@@ -199,17 +219,42 @@ impl Shared {
         self.0.destroy_pending.borrow_mut().push_back(id);
     }
 
-    // Set the event callback to use for the event loop runner
-    // This the event callback is a fairly thin layer over the user-provided callback that closes
-    // over a RootActiveEventLoop reference
-    pub(crate) fn set_listener(&self, event_handler: Box<EventHandler>) {
-        {
+    pub(crate) fn start(&self, event_handler: Box<EventHandler>) {
+        let start = {
             let mut runner = self.0.runner.borrow_mut();
             assert!(matches!(*runner, RunnerEnum::Pending));
-            *runner = RunnerEnum::Running(Runner::new(event_handler));
+            if self.0.monitor.is_initializing() {
+                *runner = RunnerEnum::Initializing(Runner::new(event_handler));
+                false
+            } else {
+                *runner = RunnerEnum::Running(Runner::new(event_handler));
+                true
+            }
+        };
+
+        if start {
+            self.init();
+            self.set_listener();
         }
```

Comment:
Just looking into it again, can't we just inline the `if start` into the branch with `true`? Or do you implicitly rely on `runner` borrow being dropped, if so I'd probably document that?

Reply:
Yes, `runner` has to be dropped, that's the reason for this whole block.

Reply:
I removed the whole block by just calling `drop(runner)`.